### ISSUE TITLE
Use Cases: refactor

### ISSUE THREAD

**Issue by aguschin** [iterative/mlem.ai#192]
From discussion with @shcheklein. TLDR:
- use cases should high-level and answer to common scenarios users have in mind. Such as MR, Productionization, or Deployment to preview environments (should be formulated differently though).
- if we would have enough time, we could cover "[five patterns to put the ML model in production: Model-as-Service, Model-as-Dependency, Precompute, Model-on-Demand, and Hybrid-Serving](https://ml-ops.org/content/three-levels-of-ml-software#code-deployment-pipelines)." 
- CI/CD should be moved to User Guide (probably)
- DVC+MLEM should be [moved to User Guide](https://github.com/iterative/mlem.ai/issues/138) (definitely)

Overall, since the team is small and docs are big, we need to thing what we can throw away - it slows us down. We need to work on simple things and GS adoption instead of covering extra/corner scenarios in docs.

**Comment by jorgeorpinel** [comment]
> use cases should high-level and answer to common scenarios users have in mind

Yup. Think problem/solution stories for ML team leads or even managers who don't want to see code or command line yet. We tend to use basic diagrams instead, and general language in the text (as little technical as possible).

From the existing pages:

- [Versioning MLEM objects with DVC](https://mlem.ai/doc/use-cases/dvc): this is more of a general guide (DVC integration). Versioning itself is not really a use case of MLEM, I think.

- [Machine Learning Model Registry](https://mlem.ai/doc/use-cases/model-registry) & [Pure Mlem Model Registry](https://mlem.ai/doc/use-cases/mlem-mr): The former is a proper use case but non-MLEM specific. The latter could be somehow merged in (without all the instructions) so it belongs better in mlem.ai .
  > We could even remove the former from dvc.org at that point (it's a bit stretched to say that its a UC of DVC).

- [CI/CD for Machine Learning](https://mlem.ai/doc/use-cases/cicd): the intro is good although I don't think this is really about CI/CD (more about distribution?) -- we can work on it 🙂 

> MR, Productionization, or Deployment to preview environments

I think def. productization is the highest priority one we should work on next though, maybe in parallel with doing some of the changes I outlined above.

> cover "five patterns to put the ML model in production

Sounds more like a blog post to me.

>  CI/CD should be moved to User Guide (probably)

Yes for the bottom sections with instructions, or maybe even a blog since it's mostly about Github actions (we don't want to document 3rd party tools).

**Comment by jorgeorpinel** [comment]
> Marking p1 since good UCs are expected to help drive traffic and adoption to the tool. Note that however I still think the GS is a higher priority atm.
